### PR TITLE
add csv data model to unit tests

### PR DESCRIPTION
**Pull request by lakikowolfe**
Include testing of csv data model in addition to jsonld

Related Jira ticket: https://sagebionetworks.jira.com/browse/FDS-1132

### CODE REVIEW

**Review by GiaJordan — CHANGES_REQUESTED** [review]

**Review by GiaJordan — CHANGES_REQUESTED** [review]
Looking through this again, it seems like the tests in `test_manifest.py` and `test_validation.py` are still only using the `.jsonld` could you update to test with the CSVs where the schemas module is concerned as well?

**Comment by linglp** [comment]
@lakikowolfe can ticket be linked to PR description? Also, when I queried the logs of `manifest/generate`, I didn't see any projects that are using .csv url when generating a manifest.. I am curious where the "ask" comes from...

**Comment by lakikowolfe** [comment]
@linglp this is the ticket: https://sagebionetworks.jira.com/browse/FDS-1132

I believe the motivation behind this ticket is to update the tests to include csv because it is a supported file type and we'd want to know if for some reason csv input was not working.

**Comment by lakikowolfe** [comment]
@GiaJordan My understanding is that this is low priority so I am focusing on the HTAN bug ticket that I have rn and will circle back to this when I can. Please let me know if I am misunderstanding the priority of this ticket and we can discuss.

**Comment by GiaJordan** [comment]
@lakikowolfe yeah your understanding is correct

**Comment by GiaJordan** [comment]
@lakikowolfe yeah from my perspective your understanding is correct, but I'll loop in kevin so he can make the final determination

**Comment by GiaJordan** [comment]
~~Looking through this again, it seems like the tests in test_manifest.py and test_validation.py are still only using the .jsonld could you update to test with the CSVs where the schemas module is concerned as well?~~
@lakikowolfe sorry about this earlier comment, looking at the current `test_schemas.py` I see it's already using both the csv and jsonld. you can disregard

**Review by GiaJordan — APPROVED** [review]

**Comment by GiaJordan** [comment]
Approving pending resolution of merge conflicts

File: tests/test_api.py
Comment (thomasyu888) [inline tests/test_api.py:324]: This is neat @lakikowolfe , just for my education, does this mean that the tests run twice for both data_model_urls?

File: tests/test_api.py
Comment (thomasyu888) [inline tests/test_api.py:78]: @lakikowolfe Thanks for the great work here!  As noted below, I really like this solution.  I just wanted to understand your rationale behind  why you chose this over adding these to `pytest.parametrize`

**Review by thomasyu888 — APPROVED** [review]
Great review @GiaJordan , and thanks for the work @lakikowolfe .  Please answer some  of my comments, but I'm going to go ahead and approve.

File: tests/test_api.py
Comment (lakikowolfe) [reply tests/test_api.py:324]: Yep that's correct

File: tests/test_api.py
Comment (lakikowolfe) [reply tests/test_api.py:78]: hmm I don't have a rationale except for when I was looking into fixtures (I'm new to pytest) I came across this method. I don't know the difference between the two but will look into it tomorrow.

Comment (thomasyu888) [reply tests/test_api.py:78]: No worries, a quick browse of this documentation: https://docs.pytest.org/en/7.1.x/how-to/parametrize.html and some of the tests below may provide some guidance. 

That said, this was more of a self-learning exercise.  A quick summary is that it would've required more code changes, but I'll leave it to you to see why.

**Pull request merged**
